### PR TITLE
inset: Add aliases for R and J and fix two examples

### LINKED ISSUE #1930: Wrong point location in inset
**Description of the problem**

When plotting a point in an inset, it is drawn in the wrong geographic location.

**Full code that generated the error**

```python
import pygmt

region = [-117.628, -117.478, 35.605, 35.801]
fig = pygmt.Figure()
pygmt.makecpt(
    cmap='etopo1',
    series='0/2800/200',
    continuous=True
)
topo_data = "@earth_relief_01s"
fig.grdimage(
    grid=topo_data,
    region=region,
    projection='M4i',
    shading=True,
    frame=True
)

with fig.inset(position="jTR+w3c", box="+pblack"):
    # Use a plotting function to create a figure inside the inset
    fig.coast(
        region=[-125, -110, 30, 45],
        projection="M3c",
        land="gray",
        borders=[1, 2],
        shorelines="1/thin",
        water="white"
    )
    fig.plot(x=-117.55, y=35.7, size=[0.2], style="c", color="black")
fig.show()
```
**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

PyGMT information:
  version: v0.6.0
System information:
  python: 3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)]
  executable: C:\Users\eyals\AppData\Local\Programs\Python\Python310\python.exe
  machine: Windows-10-10.0.19044-SP0
Dependency information:
  numpy: 1.21.2
  pandas: 1.3.3
  xarray: 0.19.0
  netCDF4: 1.5.7
  packaging: 21.0
  ghostscript: 9.54.0
  gmt: 6.3.0
GMT library information:
  binary dir: C:/Users/eyals/AppData/Local/Programs/Python/Python310
  cores: 4
  grid layout: rows
  library path: c:/programs/gmt6/bin/gmt_w64.dll
  padding: 2
  plugin dir: c:/programs/gmt6/bin/gmt_plugins
  share dir: c:/programs/gmt6/share
  version: 6.3.0

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Fix two inset examples as pointed out in https://github.com/GenericMappingTools/pygmt/issues/1930#issuecomment-1134913236.

Closes #1930.

### CODE REVIEW

**Review by willschlitzer — APPROVED** [review]
Good improvement to this tutorial

File: examples/tutorials/advanced/insets.py [hunk 1]
```diff
@@ -97,11 +97,13 @@
     frame="a",
 )
 # This does not include an inset fill as it is covered by the inset figure
-with fig.inset(position="jBL+w3c+o0.5c/0.2c", box="+pblack"):
-    # Use a plotting function to create a figure inside the inset
+with fig.inset(position="jBL+w3c/4c+o0.5c/0.2c", box="+pblack"):
+    # Use a plotting function to create a figure inside the inset.
+    # Use '?' when specify the projection width, to automatically
```
Comment (michaelgrund) [inline examples/tutorials/advanced/insets.py:102]:
```suggestion
    # Use '?' when specifying the projection width, to automatically
```

File: pygmt/src/inset.py [hunk 2]
```diff
@@ -110,10 +110,12 @@ def inset(self, **kwargs):
     >>> # Use a "with" statement to initialize the inset context manager
     >>> # Setting the position to top left and a width of 3.5 centimeters
     >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
-    ...     # Map elements under the "with" statement are plotted in the inset
+    ...     # Map elements under the "with" statement are plotted in the inset.
+    ...     # Use '?' when specify the projection width, to automatically
```
Comment (michaelgrund) [inline pygmt/src/inset.py:114]:
```suggestion
    ...     # Use '?' when specifying the projection width, to automatically
```

**Review by michaelgrund — APPROVED** [review]

**Comment by seisman** [comment]
I mark the PR as a draft due to the concern/issue mentioned in https://github.com/GenericMappingTools/pygmt/issues/1930#issuecomment-1135452245

**Comment by seisman** [comment]
As explained in https://github.com/GenericMappingTools/gmt/pull/6751#issuecomment-1138943215, a better solution is specifying -R and -J in **inset begin** so that **inset** can determine the inset width and height automatically. I think this is an intuitive solution and should be recommended, so I have updated the two examples accoordingly.

Unfortunately, **-R** and **-J** are not aliased in PyGMT v0.6.1, so this PR also adds the aliases for **-R** and **-J**.

Some TODOs:

- [ ] Update https://github.com/GenericMappingTools/pygmt/blob/main/examples/gallery/embellishments/inset.py
- [ ] Update https://github.com/GenericMappingTools/pygmt/blob/main/examples/gallery/embellishments/inset_rectangle_region.py
- [ ] Remove the inline example for inset (https://github.com/GenericMappingTools/pygmt/blob/8ddd11ce7314e1dba34fbfa428204b54e6fd15c3/pygmt/src/inset.py) because we already have a very similar example in the [gallery](https://www.pygmt.org/dev/gallery/embellishments/inset.html#sphx-glr-gallery-embellishments-inset-py) and the gallery image is shown [at the bottom of the API documentation page](https://www.pygmt.org/dev/api/generated/pygmt.Figure.inset.html#examples-using-pygmt-figure-inset).

Please let me know what you think before I start to work on them.

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**